pkg/controllers/deprovisioning/types.go

```diff
@@ -136,38 +136,30 @@ func (c *Candidate) lifetimeRemaining(clock clock.Clock) float64 {
 	return remaining
 }
 
-type action byte
+type Command struct {
+	candidates   []*Candidate
+	replacements []*scheduling.Machine
+}
 
 const (
-	actionDelete action = iota
-	actionReplace
-	actionDoNothing
+	noOpAction = "no-op"
+	replaceAction = "replace"
+	deleteAction = "delete"
 )
 
-func (a action) String() string {
-	switch a {
-	// Deprovisioning action with no replacement machines
-	case actionDelete:
-		return "delete"
-	// Deprovisioning action with replacement machines
-	case actionReplace:
-		return "replace"
-	case actionDoNothing:
-		return "do nothing"
-	default:
-		return fmt.Sprintf("unknown (%d)", a)
+func (o Command) action() string {
```

Comment:
It probably makes sense that this should be a public method on Command

Reply:
Went back and forth on this, decided against exporting since it wasn't being used elsewhere, but I'll change it back to making it public 👍